### PR TITLE
Change mrtrix mamba env to compile from github tag 3.0.4

### PR DESCRIPTION
**Pull request by RicardoRios46**
Compiles mrtrix from github tag 3.0.4 instead of mamba enviroment (that is not properly working for ubuntu 24.04).

### CODE REVIEW

File: roles/science/tasks/mrtrix.yml [hunk 1]
```diff
@@ -37,6 +16,35 @@
...
       - libpng-dev
 
+
+- name: Create directories
+  file:
+    path: "{{ item }}"
+    state: directory
+  loop:
+    - /opt/quarantine/software/mrtrix3/3.0.4
+    - /opt/quarantine/software/mrtrix3tissue/5.2.9
```
Comment (gdevenyi) [inline roles/science/tasks/mrtrix.yml:26]: can you adjust this so its not moved from the top.

I have plans in the future to make ansible functions to do directory creation etc and I'd like to keep the plays as uuniform as possible

File: roles/science/tasks/mrtrix.yml [hunk 2]
```diff
@@ -37,6 +16,35 @@
...
+  ansible.builtin.git:
+    repo: https://github.com/MRtrix3/mrtrix3
+    dest: /opt/quarantine/software/mrtrix3/3.0.4/install
+    version: 3.0.4
+
+- name: Build mrtrix3
+  shell: ./configure && ./build
+  args:
+    creates: /opt/quarantine/software/mrtrix3/3.0.4/install/bin/mrview
+    chdir: /opt/quarantine/software/mrtrix3/3.0.4/install
```
Comment (gdevenyi) [inline roles/science/tasks/mrtrix.yml:39]: well that's surprisingly straightforward

File: roles/science/tasks/mrtrix.yml [hunk 3]
```diff
@@ -37,6 +16,35 @@
       - libtiff5-dev
       - libpng-dev
```
Comment (gdevenyi) [inline roles/science/tasks/mrtrix.yml:17]: lets add git here since its needed for the clone

We also need the complete list of dependencies here:
https://mrtrix.readthedocs.io/en/3.0.4/installation/build_from_source.html#linux

This kind of dev needs to be tested on a empty machine (VM) so that we make sure its not "your system" that happens to already have everything it needs.

**Review by gdevenyi — CHANGES_REQUESTED** [review]

File: roles/science/tasks/mrtrix.yml (hunk 3 above)
Comment (RicardoRios46) [reply roles/science/tasks/mrtrix.yml:17]: Ok. I think we were missing git, g++ and python-is-python3 form the list.  python-is-python3 will be installed with afni anyways but I'll add it here too.

Comment (RicardoRios46) [reply roles/science/tasks/mrtrix.yml:17]: I'm gonna test the changes in a fresh VM only using mrtrix3 tag.

**Comment by RicardoRios46** [comment]
I tested latest changeds in a fresh ubuntu 24.04 VM (only mrtrix and modules tags) and looks good.

**Comment by RicardoRios46** [comment]
Side note: mrtrix and mrtrixtissue share a lot of binaries. Maybe their modules files should have a conflict for each other.

**Comment by RicardoRios46** [comment]
Checking your comments for #48 I understood how group_vars works. I'm going do it here to also have mrtrix 3.0.4 version in the 'all' file.

**Comment by RicardoRios46** [comment]
> Checking your comments for #48 I understood how group_vars works. I'm going do it here to also have mrtrix 3.0.4 version in the 'all' file.

Ok I was playing with this in https://github.com/RicardoRios46/NeuroAnsible/tree/mrtrix3_group_vars and it works well for 3.0.4. 

Buuut compilation fails for 3.0.3. Probably missing dependancies and stuff. Maybe its better if the version is hard coded? hehe

Anyways if you are okey with having mrtrix version in groups_vars I can merge that branch into this one.

**Comment by RicardoRios46** [comment]
Ok! Latest commits have groups_vars changes.

File: roles/science/tasks/mrtrix.yml [hunk 4]
```diff
@@ -4,30 +4,19 @@
...
-- name: Install module
-  template:
-    src: templates/conda_module.j2
-    dest: /opt/quarantine/software/mrtrix3/3.0.4/module
-  vars:
-    modulename: mrtrix3
-  notify: link modules
 
-- name: Install mtrix3tissue dependencies
+- name: Install mtrix3 dependencies
```
Comment (gdevenyi) [inline roles/science/tasks/mrtrix.yml:11]: typo

Comment (RicardoRios46) [reply roles/science/tasks/mrtrix.yml]: I also noticed the typo in the branch name too late hehe

**Review by gdevenyi — APPROVED** [review]
